drivers: usb: stm32: HAL return value handling

etienne-lms opened the pull request:
Add missing test of some HAL functions return value.

etienne-lms committed:
drivers: usb: udc: stm32: test HAL return value

Add missing test of some HAL functions return value.

Signed-off-by: Etienne Carriere <etienne.carriere@st.com>

etienne-lms commented on drivers/usb/udc/udc_stm32.c:
```diff
@@ -518,7 +536,11 @@ static void handle_msg_setup(struct udc_stm32_data *priv)
-		HAL_PCD_SetAddress(&priv->pcd, setup->wValue);
+		hal_ret = HAL_PCD_SetAddress(&priv->pcd, setup->wValue);
+		if (hal_ret != HAL_OK) {
+			LOG_ERR("HAL_PCD_SetAddress() fialed: %d", hal_ret);
```
Thanks

etienne-lms committed:
drivers: usb: udc: stm32: remove useless OR operation on return value

Remove useless OR operation on return value is udc_stm32_enable()
since the return value is always 0 before it's OR-ed with a new return
value.

No functional changes.

Signed-off-by: Etienne Carriere <etienne.carriere@st.com>

etienne-lms commented:
Rebased.

erwango commented on drivers/usb/udc/udc_stm32.c:
```diff
@@ -705,7 +705,7 @@ int udc_stm32_init(const struct device *dev)
 	HAL_StatusTypeDef status;
 
-	if (priv->clk_enable && priv->clk_enable()) {
+	if ((priv->clk_enable != NULL) && priv->clk_enable() != 0) {
```
```suggestion
	if ((priv->clk_enable != NULL) && (priv->clk_enable() != 0)) {
```

etienne-lms commented:
Thanks for the review. Comments addressed + added a few parentheses pairs for consistency.

jfischer-no commented on drivers/usb/udc/udc_stm32.c:
```diff
@@ -211,15 +211,15 @@ void HAL_PCD_ResetCallback(PCD_HandleTypeDef *hpcd)
 	/* Re-Enable control endpoints */
 	ep = udc_get_ep_cfg(dev, USB_CONTROL_EP_OUT);
-	if (ep && ep->stat.enabled) {
+	if ((ep != NULL) && ep->stat.enabled) {
```
It would be better to rename `struct udc_ep_config *ep;` to struct `udc_ep_config *ep_cfg;` or `struct udc_ep_config *cfg` because ep is usually used for endpoint address.

Then I am fine with `if (ep_cfg != NULL && ep->stat.enabled) {` but not with unnecessary parentheses like `if ((ep != NULL) && ep->stat.enabled) {`.

etienne-lms replied:
OK.

etienne-lms committed:
drivers: usb: udc: stm32: test essentially boolean value

Change implementation to ensure all boolean test are done on essentially
boolean values.

No functional changes.

Signed-off-by: Etienne Carriere <etienne.carriere@st.com>

etienne-lms committed:
drivers: usb: udc: stm32: inline comment reference to selected_speed

Clarify inline comment referring to field selected_speed of
struct udc_stm32_config.

Signed-off-by: Etienne Carriere <etienne.carriere@st.com>

etienne-lms committed:
drivers: usb: udc: stm32: rename endpoint config variable

Rename endpoint configuration pointer variable in HAL_PCD_ResetCallback(),
udc_stm32_ep_mem_config() and udc_stm32_ep_disable() from ep to ep_cfg
to prevent confusion since ep is usually used as endpoint address.

For consistency, also use ep_cfg elsewhere where an endpoint config
pointer is used, instead of using either cfg, ecfg and ep_cfg.

No functional changes.

Suggested-by: Johann Fischer <johann.fischer@nordicsemi.no>
Signed-off-by: Etienne Carriere <etienne.carriere@st.com>

mathieuchopstm commented on drivers/usb/udc/udc_stm32.c:
```diff
@@ -1170,7 +1170,7 @@ static enum udc_bus_speed udc_stm32_device_speed(const struct device *dev)
 	 * N.B.: pcd.Init.speed is used here on purpose instead
-	 * of cfg->selected_speed because HAL updates this field
+	 * of udc_stm32_config::selected_speed because HAL updates this field
 	 * after USB enumeration to reflect actual bus speed.
```
Nit:
```diff
	 * N.B.: pcd.Init.speed is used here on purpose instead
-	 * of udc_stm32_config::selected_speed because HAL updates this field
-	 * after USB enumeration to reflect actual bus speed.
+	 * of udc_stm32_config::selected_speed because HAL updates
+	 *  this field after USB enumeration to reflect actual bus speed.
```

mathieuchopstm reviewed (commented):
otherwise LGTM

etienne-lms commented on drivers/usb/udc/udc_stm32.c:
```diff
@@ -634,7 +634,7 @@ static void handle_msg_setup(struct udc_stm32_data *priv)
 	udc_ctrl_update_stage(dev, buf);
 
-	if (!buf->len) {
+	if (buf->len == 0) {
```
Indeed :-)

etienne-lms committed:
drivers: usb: udc: stm32: remove useless test on buffer length

Remove useless test on buffer length in handle_msg_setup() since the
buffer was allocated few lines above with a known non zero size.

Signed-off-by: Etienne Carriere <etienne.carriere@st.com>

etienne-lms committed:
drivers: usb: udc: stm32: split variable definition in multi-line

Split variables definition into 2 line as per Zephyr coding style
expectations since they are of different type.

No functional changes.

Signed-off-by: Etienne Carriere <etienne.carriere@st.com>

etienne-lms commented:
Comment addressed and typo (reason for CI build failures) fixed.

etienne-lms commented:
Updated.
